cms/djangoapps/contentstore/utils.py

```diff
@@ -396,16 +396,34 @@ def get_settings_url_for_category(cls, category, child, parent):
             - Problems: Unit url
         """
         handler = category
+        url = '//' + settings.CMS_BASE
+
+        if category == 'chapter':
+            url += '/course/'
+            url += unicode(parent.id)
+            url += '?show='
+            url += unicode(child.location)
+            return url
+
+        elif category == 'sequential':
+            url += '/course/'
+            url += unicode(child.location.course_key)
+            url += '?show='
+            url += unicode(child.location)
+            return url
+
+        elif category == 'vertical':
+            url += '/container/'
+            url += unicode(child.location)
+            return url
 
-        if category == "chapter":
-            usage_key_string = unicode(parent.id)
-        elif category == "sequential" or category == "unit":
-            usage_key_string = unicode(child.location)
         else:
-            usage_key_string = unicode(parent.location)
-            handler = 'unit'
-
-        return reverse('component_handler', kwargs={'usage_key_string': usage_key_string, 'handler': handler})
+            assertEqual(handler, 'problem')
```

Comment:
I thought it would add a little extra security and clarity - the original code set any 'category' or 'handler' that wasn't a 'chapter', 'sequential', or 'unit' to be a 'unit'.  Unfortunately, this meant that anything else, including 'problem' objects, would be handled in the else statement, even though they were returned as 'unit' objects.  I thought it was confusing, and tried to change things in the second commit of this pull request.  I hoped that this code would make sure that any object that failed the if and elif statements was a 'problem' and not something weird.

If this is unnecessary, I can remove this second commit right away!